### ISSUE TITLE
4Tb USB Disk formatted at 2Tb

### ISSUE THREAD

**Issue by ik0api** [syncloud/platform#522]
When I request formatting a 4Tb USB external disk Syncloud announces it will format at 2Tb. And it does it, leaving 1,7 unused and unpartitioned.

Tried with a WD Elements and a Maxtor M3

![immagine](https://user-images.githubusercontent.com/66923922/84604860-50f9d100-ae99-11ea-9157-590ab90159e0.png)

In the past I was able to format at 4Tb regularly, with the same RPI4 4Gb

Regards

**Comment by cyberb** [comment]
This is not good and it is related to MBR (32bit) limitation and we should switch to GPT (64bit).
https://en.wikipedia.org/wiki/GUID_Partition_Table#History

**Comment by cyberb** [comment]
There is a test version, could you try?
```
snap refresh platform --channel=master
```
If you format the disk it should take the full capacity.
Obviously you will lose all the data on it!

**Comment by ik0api** [comment]
mmh... I have very little experience with Linux, that's why I'm a happy user of the great Syncloud!
I would format a new disk at 4Tb then transfer the folders of the now-2Tb one. Would it work?

**Comment by cyberb** [comment]
Yes

**Comment by cyberb** [comment]
Have you tried?

**Comment by ik0api** [comment]
Not yet.
I'm very basic on Linux, I need to learn how to move files between disks
and then I'll try

At 14:39 21/06/2020, Boris Rybalkin wrote:
Have you tried?
­
You are receiving this because you authored the thread.
Reply to this email directly,

view it on GitHub, or

unsubscribe.

**Comment by cyberb** [comment]
Update released, reopen if it is not working.

**Comment by ik0api** [comment]
> 
> 
> There is a test version, could you try?
> 
> ```
> snap refresh platform --channel=master
> ```
> 
> If you format the disk it should take the full capacity.
> Obviously you will lose all the data on it!

I successfully upgraded to a 12Tb WD USB while keeping all my files and settings!

Firstly, I saved an image of the 2Tb partition with Aomei Backupper Free (for Windows), which can handle EXT4. Also for backup.
Then I cloned the 2Tb MBR partition to the new 12Tb disk (originally NTFS) with Aomei Backupper Free
Then I connected the 12Tb disk to an RPI2 with Noobs (Debian), installed Gparted and expanded the partition from 2Tb to the full 12 Tb space. Gparted automatically converted to GPT, I think.

I connected the new disk to my Syncloud RPI4 and now have all my files, all apps and all settings running perfectly.

Note: I didn't set Nextcloud to encrypt my files, don't know wether this can trouble the HD switch.
